test/e2e/azure_privatecluster.go

```diff
@@ -160,25 +160,20 @@ func AzurePrivateClusterSpec(ctx context.Context, inputGetter func() AzurePrivat
 		// cluster is deleted at the end of the test. If we don't delete this cluster, the Azure resource delete
 		// verification will fail.
 		if !input.SkipCleanup {
```

Comment:
since we're passing in the skip cleanup value to the cleanupInput now, we can remove this if statement so we always collect resources/logs regardless of skipCleanup value but only delete the cluster if skipCleanup is true